addons/website/controllers/main.py

```diff
@@ -835,56 +835,112 @@ def seo_suggest(self, keywords=None, lang=None):
-                        "res_id": model['id'],
-                        "id": f"{model['model']}-{model['id']}-{index}",
-                        "field": model.get('field'),
-                    })
+            media_elements = tree.xpath('//img[@src] | //iframe')
+            for index, el in enumerate(media_elements):
+                tag = el.tag.lower()
+                if tag == 'img':
+                    role = el.get('role')
+                    decorative = role == 'presentation'
+                    alt = el.get('alt')
+                    if not decorative or alt is None:
+                        result.append({
+                            'type': 'img',
+                            'src': el.get('src'),
+                            'alt': alt or '',
+                            'decorative': False,
+                            'updated': False,
+                            'res_model': model['model'],
+                            'res_id': model['id'],
+                            'id': f"{model['model']}-{model['id']}-img-{index}",
+                            'field': field,
+                        })
+                elif tag == 'iframe':
+                    title = el.get('title')
+                    hidden = el.get('aria-hidden') == 'true'
+                    decorative = hidden and not title
+                    src = el.get('src')
+                    if not decorative or title is None:
+                        result.append({
+                            'type': 'video',
+                            'src': src,
+                            'title': title or '',
+                            'thumbnail_url': self.extract_thumbnail_url(src),
+                            'decorative': False,
+                            'updated': False,
+                            'res_model': model['model'],
+                            'res_id': model['id'],
+                            'id': f"{model['model']}-{model['id']}-video-{index}",
+                            'field': field,
+                        })
         return json.dumps(result)
 
-    @http.route(['/website/update_alt_images'], type='jsonrpc', auth="user", website=True)
-    def update_alt_images(self, imgs):
+    @http.route(['/website/update_media_elements'], type='jsonrpc', auth="user", website=True)
+    def update_media_elements(self, elements):
         if not request.env.user.has_group('website.group_website_restricted_editor'):
             raise werkzeug.exceptions.Forbidden()
-        for img in imgs:
-            record = request.env[img['res_model']].browse(img['res_id'])
+        # Group updates by (model, id, field)
+        updates_by_record = {}
+        for el in elements:
+            key = (el['res_model'], el['res_id'], el['field'])
+            updates_by_record.setdefault(key, []).append(el)
+
+        for (model, res_id, field), updates in updates_by_record.items():
+            record = request.env[model].browse(res_id)
```

Comment:
Why all this stuff? why not simply iterate on elements and inside loop get the res_model, res_id and field?